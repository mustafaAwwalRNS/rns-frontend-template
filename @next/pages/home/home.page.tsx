import React from "react";
import { GuestPanel } from "@components/templates";
import { Typography, Divider, Container } from "@components/atoms";

/**
 *
 * Home
 *
 * Features:
 * - Render home page for the guest users.
 */
export const Home = (): JSX.Element => {
  return (
    <Container spacing={2}>
      <Typography variant="h1">Welcome to RNS front end template</Typography>
      <Divider />
      <Typography>
        This front end template is created with extreme care and study.
      </Typography>
    </Container>
  );
};

Home.getLayout = (page: React.ReactNode) => <GuestPanel>{page}</GuestPanel>;
