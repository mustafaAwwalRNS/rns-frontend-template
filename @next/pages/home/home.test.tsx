import React from "react";
import { render } from "test-utils";
import { Home } from "./home.page";
const setup = () => {
  const { container } = render(<Home />);
  return container;
};
describe("<Home />", () => {
  it("should match snapshot", () => {
    const home = setup();
    expect(home).toMatchSnapshot();
  });
});
