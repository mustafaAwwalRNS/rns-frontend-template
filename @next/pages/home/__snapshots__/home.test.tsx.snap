// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Home /> should match snapshot 1`] = `
.c1 {
  margin: 0;
  padding: 0;
  color: black;
}

.c3 {
  margin: 0;
  padding: 0;
  color: black;
}

.c2 {
  border-top: 1px solid #ccc;
}

.c0 {
  max-width: 1320px;
  display: block;
  margin: 0 auto;
  padding: 1rem;
}

<div>
  <div
    class="c0"
    spacing="2"
  >
    <h1
      class="c1"
    >
      Welcome to RNS front end template
    </h1>
    <hr
      class="c2"
    />
    <span
      class="c3"
    >
      This front end template is created with extreme care and study.
    </span>
  </div>
</div>
`;
